src/app/services/student-round.service.ts

```diff
@@ -145,14 +145,61 @@ export class StudentRoundService {
     shareReplay(1),
   );
 
+  interactions$: Observable<Interaction[]> =
+  combineLatest([this.sessionService.currentRoundPath$, this.authService.currentUser$]).pipe(
+    switchMap(([path, user]) =>
+      path && user
+        ? this.firebaseService.getStudentInteractions(path, user.uid)
+        : of([]))
+  );
+
+  totalPollen$: Observable<number> = this.interactions$.pipe(
+    map(interactions =>
+      interactions.filter(interaction => !interaction.isNest)
+    ),
+    map(interactions =>
+      interactions.length)
+  );
+
+  /**
+   * An observable that emits the amount of pollen a bee is currently carrying
+   * In order to do so it requires the array of interactions to be sorted from most recent to least recent
+   */
+  currentBeePollen$: Observable<number> =
+  this.interactions$.pipe(
+    map(interactions => {
+        let currentPollen = 0;
+        for (const interaction of interactions) {
+          if (interaction.isNest) {
+            break;
+          }
+          currentPollen++;
+        }
+        return currentPollen;
+    }),
+    distinctUntilChanged(),
+  );
+
+  currentNestPollen$: Observable<number> = combineLatest([this.totalPollen$, this.currentBeePollen$]).pipe(
+    map(([totalPollen, beePollen]) =>
+      totalPollen - beePollen
+    ),
+    distinctUntilChanged(),
+    shareReplay(1)
+  );
+
   /**
    * Records an interaction with a specific barcode value.
    *
    * @param barcodeValue the barcode value to submit in the interaction
+   * @param isANest whether the barcode corresponds to a student's nest
    */
-  interact(barcodeValue: number) {
+  interact(barcodeValue: number, isANest?: boolean) {
+    let isNest: boolean;
+    isANest ? isNest = true : isNest = false;
```

Comment:
What's going on here? I assume you wanted to allow `isANest` to be optional, in that case you can just set a default value.
```suggestion
  interact(barcodeValue: number, isNest: boolean = false) {
```

But I don't think it even has to be an optional thing since `interact` is only called in one place that is already providing the boolean.

Reply:
It was meant to act as sort of a default value, I just didn't realize you could simply set one like that. This implementation is definitely a lot cleaner.